content/automated-rust-releases.md

```diff
@@ -0,0 +1,582 @@
+
+```sh
+# Changelog
+
+All notable changes to this project will be documented in this file.
+
+## [unreleased]
+
+### Documentation
+
+- Add link to emacs package support git-cliff (#307)
+```
+
+For this configuration, first we should install `git-cliff`:
+
+```sh
+cargo install git-cliff
+```
+
+Then we can initialize it as follows:
+
+```sh
+git cliff --init
+```
+
+This will create `cliff.toml` in the current directory and you can commit it in the root directory of your repo for `release-plz` to pick it up.
+
+If you want to generate a fancier changelog, you can look at the [examples](https://git-cliff.org/docs/templating/examples) or use the [configuration that `git-cliff` uses](https://github.com/orhun/git-cliff/blob/main/cliff.toml) which corresponds to the following:
+
+```sh
+$ git cliff
+
+# Changelog
+
+All notable changes to this project will be documented in this file.
+
+## [unreleased]
+
+### 📚 Documentation
+
+- _(readme)_ Add link to emacs package support git-cliff ([#307](https://github.com/orhun/git-cliff/issues/307)) - ([fa471c7](https://github.com/orhun/git-cliff/commit/fa471c7178dce184ca6fe5bbb24b9c2db96d68ce))
+```
+
+Great, now we are ready to set up the actual releaser tool!
+
+---
+
+### **release-plz**
+
+> [release-plz](https://release-plz.ieni.dev) creates a fully automated release pipeline, allowing you to easily release changes more frequently, without the fear of doing typos or other subtle manual mistakes you can make when releasing from your terminal.
+
+Unlike `git-cliff`, we don't actually need to install `release-plz` as a command-line tool. It is because `release-plz` runs from the CI (GitHub Actions) and we are not doing anything locally for the releases. However, we still need to tweak some things regarding how we want to release our project.
+
+For configuring `release-plz`, create `release-plz.toml` with the following contents at the root of your repository:
+
+```toml
+[workspace]
+# path of the git-cliff configuration
+changelog_config = "cliff.toml"
+
+# enable changelog updates
+changelog_update = true
+
+# update dependencies with `cargo update`
+dependencies_update = true
+
+# create tags for the releases
+git_tag_enable = true
+
+# disable GitHub releases
+git_release_enable = false
+
+# labels for the release PR
+pr_labels = ["release"]
+
+# disallow updating repositories with uncommitted changes
+allow_dirty = false
+
+# disallow packaging with uncommitted changes
+publish_allow_dirty = false
+
+# disable running `cargo-semver-checks`
+semver_check = false
+```
+
+The important options here are:
+
+- `changelog_config`: should point out to the `git-cliff` configuration which we created in the previous section.
+- `dependencies_update`: setting this to `true` means that you want to run `cargo update` before the each release. Enabling this for updating the transitive dependencies wouldn't hurt since we will already be updating the main dependencies with Dependabot in the future steps.
```

Comment:
```suggestion
- `dependencies_update`: setting this to `true` means that you want to run `cargo update` before each release. Enabling this for updating the transitive dependencies wouldn't hurt since we will already be updating the main dependencies with Dependabot in the future steps.
```